tests/db_functions/text/test_concat.py

```diff
@@ -75,17 +75,16 @@ def test_mixed_char_text(self):
         expected = article.title + " - " + article.text
         self.assertEqual(expected.upper(), article.title_text)
 
-    @skipUnless(connection.vendor == "sqlite", "sqlite specific implementation detail.")
-    def test_coalesce_idempotent(self):
-        pair = ConcatPair(V("a"), V("b"))
-        # Check nodes counts
-        self.assertEqual(len(list(pair.flatten())), 3)
-        self.assertEqual(
-            len(list(pair.coalesce().flatten())), 7
-        )  # + 2 Coalesce + 2 Value()
-        self.assertEqual(len(list(pair.flatten())), 3)
-
     def test_sql_generation_idempotency(self):
         qs = Article.objects.annotate(description=Concat("title", V(": "), "summary"))
         # Multiple compilations should not alter the generated query.
         self.assertEqual(str(qs.query), str(qs.all().query))
+
+    def test_concat_non_str(self):
```

Comment:
Before 09ffc5c1212d4ced58b708cbbf3dfbfb77b782ca `Concat` required that non-text expressions were cast'ed to work with concat but since this commit inadvertently changed it I figured it would be backward incompatible to change now.